test/e2e/packetcapture_test.go

```diff
@@ -742,43 +913,17 @@ func runPacketCaptureTest(t *testing.T, data *TestData, tc pcTestCase) {
 		}
 	}()
 
-	// The destination is unset or invalid, do not generate traffic as the test expects to fail.
-	if dstPodIPs != nil {
-		srcPod := tc.pc.Spec.Source.Pod.Name
-		protocol := *tc.pc.Spec.Packet.Protocol
-		server := dstPodIPs.IPv4.String()
-		if tc.ipVersion == 6 {
-			server = dstPodIPs.IPv6.String()
-		}
-		// wait for CR running.
-		_, err := data.waitForPacketCapture(t, tc.pc.Name, 0, isPacketCaptureRunning)
-		if err != nil {
-			t.Fatalf("Error: Waiting PacketCapture to Running failed: %v", err)
-		}
-		connections := 10
-		if tc.numConnections != 0 {
-			connections = tc.numConnections
-		}
-		// Send an ICMP echo packet from the source Pod to the destination.
-		switch protocol {
-		case icmpProto:
-			if err := data.RunPingCommandFromTestPod(PodInfo{srcPod, getOSString(), "", data.testNamespace},
-				data.testNamespace, dstPodIPs, toolboxContainerName, connections, 0, false); err != nil {
-				t.Logf("Ping(%s) '%s' -> '%v' failed: ERROR (%v)", protocol.StrVal, srcPod, *dstPodIPs, err)
-			}
-		case tcpProto:
-			for i := 1; i <= connections; i++ {
-				if err := data.runNetcatCommandFromTestPodWithProtocol(srcPod, data.testNamespace, toolboxContainerName, server, serverPodPort, "tcp"); err != nil {
-					t.Logf("Netcat(TCP) '%s' -> '%v' failed: ERROR (%v)", srcPod, server, err)
-				}
-			}
-		case udpProto:
-			for i := 1; i <= connections; i++ {
-				if err := data.runNetcatCommandFromTestPodWithProtocol(srcPod, data.testNamespace, toolboxContainerName, server, serverPodPort, "udp"); err != nil {
-					t.Logf("Netcat(UDP) '%s' -> '%v' failed: ERROR (%v)", srcPod, server, err)
-				}
-			}
+	// wait for CR running.
+	_, err := data.waitForPacketCapture(t, tc.pc.Name, 0, isPacketCaptureRunning)
+	if err != nil {
+		t.Fatalf("Error: Waiting PacketCapture to Running failed: %v", err)
+	}
+
+	if tc.name != "non-existing-pod" {
```

Comment:
rather than relying on the test case name in such a fashion, it's usually better to add a new field to the test case definition, e.g. `invalidDestination bool`. You only need to explicitly set it (to true) when the destination is indeed invalid, as it will default to false otherwise.